File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/Or.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.Visitable;
import org.uva.sea.ql.ast.Visitor;

public class Or extends BinaryExpr implements Visitable {
	
	public Or(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
	}
	
	@Override
	public Boolean eval() {
		return (Boolean) lhs.eval() || (Boolean) rhs.eval();
	}
	
	public void accept(Visitor visitor) {
		visitor.visit(this);
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/or.go
package binaryoperatorexpr

import (
	"ql/ast/expr"
	"ql/ast/visit"
)

type Or struct {
	Lhs, Rhs expr.Expr
}

func (o Or) GetLhs() expr.Expr {
	return o.Lhs
}

func (o Or) GetRhs() expr.Expr {
	return o.Rhs
}

func (o Or) Eval(s interface{}) interface{} {
	return o.GetLhs().Eval(s).(bool) || o.GetRhs().Eval(s).(bool)
}

func (o Or) Accept(v visit.Visitor, s interface{}) interface{} {
	return v.Visit(o, s)
}